core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java

```diff
@@ -198,30 +200,34 @@ public void testOneOwnerSuspectedNoFilter(Method m) throws ExecutionException, I
       cache(1).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new DelayingInterceptor(arrival, release1), 0);
       cache(2).getAdvancedCache().getAsyncInterceptorChain().addInterceptor(new DelayingInterceptor(arrival, release2), 0);
 
-      org.infinispan.remoting.transport.Address address1 = cache(1).getCacheManager().getAddress();
-      org.infinispan.remoting.transport.Address address2 = cache(2).getCacheManager().getAddress();
-      List<org.infinispan.remoting.transport.Address> owners = Arrays.asList(address1, address2);
+      Address address1 = address(1);
+      Address address2 = address(2);
+      List<Address> owners = Arrays.asList(address1, address2);
 
       ClusteredGetCommand clusteredGet = new ClusteredGetCommand(key, ByteString.fromString(cache(0).getName()), 0, AnyEquivalence.getInstance());
       final int timeout = 15;
       RpcOptions rpcOptions = new RpcOptions(timeout, TimeUnit.SECONDS, null, ResponseMode.WAIT_FOR_VALID_RESPONSE, DeliverOrder.NONE);
 
-      CompletableFuture<Map<org.infinispan.remoting.transport.Address, Response>> future = cache(0).getAdvancedCache().getRpcManager().invokeRemotelyAsync(owners, clusteredGet, rpcOptions);
+      RpcManager rpcManager = cache(0).getAdvancedCache().getRpcManager();
+      CompletableFuture<Map<Address, Response>> future = rpcManager.invokeRemotelyAsync(owners, clusteredGet, rpcOptions);
 
       assertTrue(arrival.await(10, TimeUnit.SECONDS));
 
       installNewView(cache(0), cache(0), cache(1));
 
+      // RequestCorrelator processes the view asynchronously, so we need to wait a bit for node 2 to be suspected
+      Thread.sleep(100);
+
       // suspection should not fail the operation
       assertFalse(future.isDone());
       long requestAllowed = System.nanoTime();
       release1.countDown();
-      Map<org.infinispan.remoting.transport.Address, Response> responses = future.get();
+      Map<Address, Response> responses = future.get();
       long requestCompleted = System.nanoTime();
-      assertTrue(responses.get(address1).isSuccessful());
-      assertNotNull(((SuccessfulResponse) responses.get(address1)).getResponseValue());
-      assertTrue(responses.get(address2) instanceof CacheNotFoundResponse);
-      assertTrue(TimeUnit.NANOSECONDS.toSeconds(requestCompleted - requestAllowed) < timeout);
+
+      assertEquals(0, TimeUnit.NANOSECONDS.toSeconds(requestCompleted - requestAllowed), timeout);
```

Comment:
This seemed the simplest way to log the actual duration after a failure, but I agree it's a bit confusing.